core/domain/question_services.py

```diff
@@ -117,6 +121,23 @@ def get_question_by_id(question_id):
         return None
 
 
+def get_questions_by_ids(question_ids):
+    """Returns a list of domain objects representing questions.
+
+    Args:
+        question_ids: list(str). List of question ids.
+
+    Returns:
+        list(Question) or None. A list of domain objects representing questions
+        with the given ids or None when the id is not valid.
+    """
+    question_model_list = question_models.QuestionModel.get_multi(question_ids)
+    questions = []
+    for question_model in question_model_list:
+        questions.append(get_question_from_model(question_model).to_dict())
```

Comment:
You can't do a to_dict() here if you actually want to return a domain object.

Reply:
Updated.